src/account/operations/transaction/input_selection.rs

```diff
@@ -225,21 +227,31 @@ fn filter_inputs(
     bech32_hrp: &str,
     outputs: &[Output],
     burn: Option<&Burn>,
+    custom_inputs: Option<&HashSet<OutputId>>,
+    mandatory_inputs: Option<&HashSet<OutputId>>,
 ) -> crate::Result<Vec<InputSigningData>> {
     let mut available_outputs_signing_data = Vec::new();
 
     for output_data in available_outputs {
-        let output_can_be_unlocked_now_and_in_future = can_output_be_unlocked_forever_from_now_on(
-            // We use the addresses with unspent outputs, because other addresses of the
-            // account without unspent outputs can't be related to this output
-            &account.addresses_with_unspent_outputs,
-            output_data,
-            current_time,
-        );
-
-        // Outputs that could get unlocked in the future will not be included
-        if !output_can_be_unlocked_now_and_in_future {
-            continue;
+        if !custom_inputs
+            .map(|inputs| inputs.contains(&output_data.output_id))
+            .unwrap_or(false)
+            && !mandatory_inputs
+                .map(|inputs| inputs.contains(&output_data.output_id))
+                .unwrap_or(false)
+        {
+            let output_can_be_unlocked_now_and_in_future = can_output_be_unlocked_forever_from_now_on(
```

Comment:
Why not `if !can_output_be_unlocked_forever_from_now_on() { continue }`?
I know it was like this, maybe for clarity sake?

Reply:
No reason, and this is all going to be refactored soon anymore